### ISSUE TITLE
Month and Year input field name attribute

### ISSUE THREAD

**Issue by pandymic** [CardJs/CardJs#13]
When doing a "manual insertion" of individual input fields as outlined in the README, the new hidden input fields that are created to handle Month and Year data do not inherit the `name` attribute. Per lines 1061-1062 the original elements are simply overwritten with hidden fields that have their names hard-coded.

This causes potential mismatch when handling input from clients who have JavaScript disabled.

One possible solution is to check if the initialized `expiryMonthInput` and `expiryYearInput` objects have a non-empty name attribute already set and simply pass it to the new hidden element.

```
if ( this.expiryMonthInput.attr("name") ) {
  // inherit the existing "name" attribute.
  this.expiryMonthInput = $("<input type='hidden' />").attr("name", this.expiryMonthInput.attr("name"));
} else {
  // just in case it hasn't already been set.
  this.expiryMonthInput = $("<input type='hidden' name='expiry-month' />");
}

if ( this.expiryYearInput.attr("name") ) {
  // inherit the existing "name" attribute.
  this.expiryYearInput = $("<input type='hidden' />").attr("name", this.expiryYearInput.attr("name"));
} else {
  // just in case it hasn't already been set.
  this.expiryYearInput = $("<input type='hidden' name='expiry-year' />");
}
```

Alternatively, since the input elements have already been initialized by the `initExpiryMonthInput` and `initExpiryYearInput` methods, just changing their type to "hidden" would also do the trick. Of course, this assumes that they're valid input elements. There's also the chance that the existence of any other attributes (`class`, `placeholder`, etc.) may cause issues down the line. I definitely like the idea of keeping hidden fields as sterile as possible.

```
this.expiryMonthInput.attr("type", "hidden");
this.expiryYearInput.attr("type", "hidden");
```

Or you could leverage the `copyAllElementAttributes` method that you've already written and have used higher up for the DROPDOWN style month and year picker.

**Comment by pandymic** [comment]
Looking back over the initialization and setup, I suspect my alternative solution appears to be the most elegant.

In this case, lines [`1061-1062`](https://github.com/CardJs/CardJs/blob/master/src/js/card-js.js#L1061-L1062) can simply be replaced with:

```
this.expiryMonthInput.attr("type", "hidden");
this.expiryYearInput.attr("type", "hidden");
```

Additionally, lines [`947`](https://github.com/CardJs/CardJs/blob/master/src/js/card-js.js#L947) and [`957`](https://github.com/CardJs/CardJs/blob/master/src/js/card-js.js#L957) should be updated to include a default "name" attribute when initializing a new DOM element.

```
this.expiryMonthInput = $("<input class='expiry-month' name='expiry-month' />");
```
```
this.expiryYearInput = $("<input class='expiry-year' name='expiry-year' />");
```


The rationale for this is quite simple. the `this.expiry*Input` variables already exist, as per the bootstrapping process that occurs when CardJS is intialized. 

It's safe to assume that these variables already exist and they contain valid jQuery objects, because they have already been set up at [initialisation](https://github.com/CardJs/CardJs/blob/master/src/js/card-js.js#L26-L27). By the time we get to `setupExpiryInput` the element is already been either created or located and detached from the DOM as per [`initExpiryMonthInput()`](https://github.com/CardJs/CardJs/blob/master/src/js/card-js.js#L942-L949) and [`initExpiryYearInput()`](https://github.com/CardJs/CardJs/blob/master/src/js/card-js.js#L952-L959).

Unless I'm making some glaring oversight, the only thing that we need to do is change the type to "hidden" in case it has not been already.

**Comment by ghenry** [comment]
Can we get this PR accepted?

**Comment by colin-stannard** [comment]
If I've understood the problem correctly, this should now be fixed as of v1.0.15 (see 05980e8)

You should be able to see a working version of this functionality with the new [example 06](https://github.com/CardJs/CardJs/blob/05980e89810a682db474eb0429da3b0fba6e5e37/examples/06_change-form-field-names.html)

I've also improved the code commenting in this area and removed a fair amount of duplication
